client/my-sites/checkout/composite-checkout/wpcom/hooks/use-shopping-cart.ts

```diff
@@ -454,6 +454,10 @@ export function useShoppingCart(
 	const addItem: ( arg0: RequestCartProduct ) => void = useCallback(
 		( requestCartProductToAdd ) => {
 			hookDispatch( { type: 'ADD_CART_ITEM', requestCartProductToAdd } );
+			onEvent?.( {
```

Comment:
I could be wrong but I'm pretty sure that this function is only used by the upsell button and I'm not sure if that's what you intended. We add items to the cart from the URL when the cart is initialized. See `useInitializeCartFromServer`.